hypershift-operator/controllers/hostedclustersizing/hostedclustersizing_controller.go

```diff
@@ -216,7 +217,9 @@ func (r *reconciler) reconcile(
 
 	// first, we figure out the node count for the hosted cluster
 	var nodeCount uint32
+	nodeCountRequiresAPIServer := false
 	if hccoReportsNodeCount {
+		nodeCountRequiresAPIServer = true
```

Comment:
should we also let the HCCO to compute nodecount from the NodePools spec?
Even without facing this issue wouldn't that make sense from the autoscaling functionallity pov?
If so we could change existing hostedControlPlane.Status.NodeCount or maybe have a new one e.g. hostedControlPlane.Status.NodeCountIntent

Reply:
Actually scratch that. In the cases where we don't have nodepools, but nodes exist (IBM Cloud), falling back on the nodepools will give us the wrong answer.